genawaiter-proc-macro/examples/await.rs

```diff
@@ -0,0 +1,81 @@
+// Remove to test otherwise CI fails because of the async closures
+#[cfg(feature = "proc_macro_example")]
```

Comment:
I don't think this needs a feature gate. Since it's an example it doesn't matter if it fails to compile without the right features. (Actually, it might be worse if it compiles but does nothing.)

Reply:
Also, I couldn't get this file to even run without moving it up a level to the root of the repo. And it seems like there's a lot going on here, maybe this would be better off as a bunch of tests?

Reply:
Yeah I was planning on probably deleting it, distributing out some test would be even better.